### PR TITLE
Use 2nd order differencing

### PR DESCRIPTION
**Pull request by matt-frey**

### CODE REVIEW

File: src/3d/inversion/inversion.f90 [hunk 1]
```diff
@@ -145,39 +68,37 @@ subroutine vor2vel(vortg,  velog,  velgradg)
             !------------------------------------------------------------
             !Compute x velocity component, u = B_z - C_y:
             call diffy(cs, ds)
-            call diffz0(bs, es, bsbot, bstop)
-            !bsbot & bstop contain spectral y vorticity component at z_min and z_max
+            call diffz0(bs, es)
             !$omp parallel
             !$omp workshare
             fs = es - ds
```
Comment (daviddritschel) [inline src/3d/inversion/inversion.f90:74]: Can we copy directly into svelog here to save an operation?  Also below

**Review by daviddritschel — COMMENTED** [review]
A few comments that may be helpful

File: src/3d/inversion/inversion.f90 (hunk 1 above)
Comment (matt-frey) [reply src/3d/inversion/inversion.f90]: Yes, we do not use `fs`.

File: src/3d/inversion/inversion_utils.f90 [hunk 2]
```diff
@@ -244,9 +222,7 @@ subroutine init_fft
             enddo
             htda(nz) = one / (one + f16 + f16 * etda(nz-2))
```
[the remark this answers is not in the document]
Comment (matt-frey) [reply src/3d/inversion/inversion_utils.f90:223]: hdta is used in the vertint routine.

File: src/3d/inversion/inversion_utils.f90 [hunk 3]
```diff
@@ -521,40 +355,38 @@ subroutine lapinv1(fs)
         !::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
 
         !Finds f by integrating df/dz = d, ensuring f = 0 at the boundaries
-        !using 4th-order compact differencing.  Here ds = df/dz and fs = f.
+        !using finite differencing.  Here ds = df/dz and fs = f.
```
[the remark this answers is not in the document]
Comment (daviddritschel) [reply src/3d/inversion/inversion_utils.f90:358]: yes, I agree!

File: src/3d/inversion/inversion_utils.f90 [hunk 4]
```diff
@@ -521,40 +355,38 @@ subroutine lapinv1(fs)
...
-            !-------------------------------------------
-            !First interpolate ds to a half grid as es:
+            ! set lower boundary value
+            fs(0)  = zero
+
+            !$omp parallel private(iz)
+            !$omp do
             do iz = 1, nz
-                es(iz) = f23 * (ds(iz-1) + ds(iz))
+                fs(iz) = fs(iz-1) + dz2 * (ds(i) - ds(i-1))
```
[the remark this answers is not in the document]
Comment (daviddritschel) [reply src/3d/inversion/inversion_utils.f90:371]: I mean + ds(i-1)

File: src/3d/inversion/inversion_utils.f90 [hunk 5]
```diff
@@ -521,40 +355,38 @@ subroutine lapinv1(fs)
...
-                es(iz) = (es(iz) - f16 * es(iz-1)) * htda(iz)
-            enddo
+            ! shift to adjust f(nz) to be zero
+            c = f(nz) / dble(nz)
 
-            do iz = nz-1, 1, -1
-                es(iz) = etda(iz) * es(iz+1) + es(iz)
+            !$omp parallel private(iz)
+            !$omp do
+            do iz = 1, nz
```
Comment (daviddritschel) [inline src/3d/inversion/inversion_utils.f90:381]: Can go to nz-1 instead of nz here as f(nz) = zero is set just below

**Review by daviddritschel — COMMENTED** [review]
Just a few small changes

File: src/3d/inversion/inversion_utils.f90 (hunk 5 above)
Comment (matt-frey) [reply src/3d/inversion/inversion_utils.f90:381]: Yes!

**Review by daviddritschel — APPROVED** [review]

**Pull request closed without merging**